src/lsptoolshost/sourceGeneratedFilesContentProvider.ts

```diff
@@ -0,0 +1,31 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as RoslynProtocol from './roslynProtocol';
+import { RoslynLanguageServer } from './roslynLanguageServer';
+import { UriConverter } from './uriConverter';
+import * as lsp from 'vscode-languageserver-protocol';
+
+export function registerSourceGeneratedFilesContentProvider(
+    context: vscode.ExtensionContext,
+    languageServer: RoslynLanguageServer
+) {
+    context.subscriptions.push(
+        vscode.workspace.registerTextDocumentContentProvider(
+            'roslyn-source-generated',
+            new (class implements vscode.TextDocumentContentProvider {
+                async provideTextDocumentContent(uri: vscode.Uri, token: vscode.CancellationToken): Promise<string> {
```

Comment:
we definitely need to add refresh support in.  There is some interesting behavior where VSCode doesn't actually always call didClose on the document even if the only active view closes (it calls sometimes much later).  So the content gets reused even if the view is closed and another go to def request is made.

I could implement a command to refresh, or refresh on textview changing, but IMHO we should just implement server side refresh instead.